src/index.d.ts

```diff
@@ -1,966 +1,54 @@
-export as namespace Reselect;
-
-export type Selector<S, R> = (state: S) => R;
-
-export type OutputSelector<S, R, C> = Selector<S, R> & {
-  resultFunc: C;
-  recomputations: () => number;
-  resetRecomputations: () => number;
-}
+// Minimum TypeScript Version: 4.2
```

Comment:
this should be 4.1, right?

Reply:
Yeah, there was a weird bug with `dtslint` that was only throwing an error in 4.1 when it shouldn't have. It literally said "change it to `4.2` to fix this problem" so I bumped it up. I am not an expert with `dtslint` thought, so there may be a more correct way to have solved that.

It did feel weird, but it was the only error when running against 4.1 and went away on 4.2 so I left it. Now that I think about this though, what would be better would be to do your earlier suggestion of splitting the types.

For the versions >=3.5,<4.0 we can use the `dtslint` set with a minimum version of 3.5 and then we can use a higher version for the "new typings".

Reply:
To bring the other conversations back here: we're going to use some form of "separated types" using `typeVersions` here, so this will no longer be necessary. We will just use the built-in `// @ts-expect-error` that was suggested instead.